### ISSUE TITLE
Animation not triggered for list of same length

### ISSUE THREAD

**Issue by francois-appliware** [angular/angular#19034]
## I'm submitting a...
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
When using animation, bound to a property, setting a transition of `* => *` to trigger every time the property changes, the animation does not trigger if the property is a list of the same length as before.

Key points are :
in list component :
```typescript
 transition('* => *', [ ... ] ) // from angular doc on stagger animation
```
```html
<div [@listAnimation]="items">
      <div *ngFor="let item of items">
        {{ item.id }} - {{ item.name }}
      </div>
</div>
```
in main component :
```html
 <my-list [items]="items | async"></my-list>
```

## Expected behavior
The animation should trigger every time the binding changes, as specified in the doc.

## Minimal reproduction of the problem with instructions
You can find a plunker here :
http://plnkr.co/edit/yL4mBLtnB4ovVf8axtl3?p=preview
How it works
- The parent component gives a list to its child
- The child displays the list items, applying an animation every time it receives a new list
- [ERROR ->] When giving a list of same size, the animation does not trigger

This is the case with List from immutable, but also applies with a simple array.

I do not know if it is the binding that does not wake up, if it is the async pipe, or something more related to the animation. It does work with a list of number though (instead of a list of object)

## What is the motivation / use case for changing the behavior?
I have an application with a master / details view. The master has a list. This list is paginated.
I want to animate the item every time the user changes the page.
I have an application with a master / details view. The master has a list of children. When the user chooses an item, the list shows its children (kind of a tree view). If same length of children, no animation.

## Environment

<pre><code>
Angular version: 4.4.0-RC.0 but same thing with 4.3.6 (I updated to see if the problem was still present)

Browser:
- Chrome (desktop) version 60

</code></pre>

**Comment by rabbitzbx** [comment]
I have a same problem, so how can I solve it?

**Comment by JoostK** [comment]
This was addressed in #19279 where `normalizeTriggerValue` no longer calls `toString()` on the value, which resulted in an identical trigger string for equally sized lists.
